packages/@aws-cdk/aws-bedrock-alpha/bedrock/agents/agent.ts

```
@@ -0,0 +1,844 @@
+        sourceAccount: Stack.of(this).account,
+      });
+    }
+
+    // ------------------------------------------------------
+    // Set Lazy Validations
+    // ------------------------------------------------------
+
+    // ------------------------------------------------------
+    // CFN Props - With Lazy support
+    // ------------------------------------------------------
+    const cfnProps: bedrock.CfnAgentProps = {
+      actionGroups: Lazy.any({ produce: () => this.renderActionGroups() }, { omitEmptyArray: true }),
+      agentName: this.name,
+      agentResourceRoleArn: this.role.roleArn,
+      autoPrepare: this.shouldPrepareAgent,
+      customerEncryptionKeyArn: props.kmsKey?.keyArn,
+      description: props.description,
+      foundationModel: this.foundationModel.invokableArn,
+      idleSessionTtlInSeconds: this.idleSessionTTL.toSeconds(),
+      instruction: props.instruction,
+      memoryConfiguration: props.memory,
+      promptOverrideConfiguration: this.promptOverrideConfiguration?._render(),
+      skipResourceInUseCheckOnDelete: this.forceDelete,
+      agentCollaboration: this.agentCollaboration,
+      agentCollaborators: Lazy.any({ produce: () => this.renderAgentCollaborators() }, { omitEmptyArray: true }),
+      customOrchestration: this.renderCustomOrchestration(),
+      orchestrationType: this.orchestrationType,
+    };
+
+    // ------------------------------------------------------
+    // L1 Instantiation
+    // ------------------------------------------------------
+    this.__resource = new bedrock.CfnAgent(this, 'Resource', cfnProps);
+
+    this.agentId = this.getResourceNameAttribute(this.__resource.attrAgentId);
+    this.agentArn = this.getResourceArnAttribute(this.__resource.attrAgentArn, {
+      service: 'bedrock',
+      resource: 'agent',
+      resourceName: this.physicalName,
+      arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
+    });
+    this.agentVersion = this.__resource.attrAgentVersion;
```

The API spec requires the following pattern, should we validate here to fail early?

`Pattern: ^(DRAFT|[0-9]{0,4}[1-9][0-9]{0,4})$
`
https://docs.aws.amazon.com/bedrock/latest/APIReference/API_agent_PrepareAgent.html#API_agent_PrepareAgent_ResponseSyntax

---

updated doc with the supported pattern - f91c1bc70d9910c1e82d5de3fb9436ad1194a77c